src/main/java/org/commonjava/util/gateway/metrics/jfr/events/JaxRSEvent.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2011-2021 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.util.gateway.metrics.jfr.events;
+
+import jdk.jfr.Category;
+import jdk.jfr.DataAmount;
+import jdk.jfr.Description;
+import jdk.jfr.Event;
+import jdk.jfr.Label;
+import jdk.jfr.Name;
+import jdk.jfr.StackTrace;
+
+@Name ( JaxRSEvent.NAME )
+@Label ( "Invocation" )
+@Category ( "JaxRS" )
+@Description ( "JaxRS invocation event" )
+@StackTrace ( false )
+public class JaxRSEvent extends Event
+{
+    public static final String NAME = "o.c.i.b.j.m.e.JaxRSEvent";
+
+    public static final ThreadLocal<JaxRSEvent> EVENT =
```

Comment:
It seems this is not used. Why is it here?

Reply:
JaxRSEvent.NAME is used by the context filter to lookup the property.
The contect filter has been updated to use JaxRSEvent.EVENT object.

Reply:
Which context filter? I guess this is a convention of jfr? because the code change in this page has no usage of JaxRSEvent.EVENT. I am asking because things get complicated on Quarkus, especially for threadlocal. With the async execution everywhere, some normal threadlocal logic breaks. I encounter threadlocal problem somewhere else. The main problem is once you set sth in threadlocal and expect to use it later, but end up with nothing because the 'later code' is executed in different thread.

Reply:
So I am expecting the same thread to be processing the request and response. If that's not the case then this proposal needs updating (am happy to do that). The context filter is the new class [FlightRecorderFilter](https://github.com/whitingjr/gateway/blob/add-jfr-events/src/main/java/org/commonjava/util/gateway/metrics/jfr/FlightRecorderFilter.java).

Reply:
Ok, I see the usage. I think you can enable the jfr, the trace log level, and run unit tests to see if logger.trace in both methods are from same thread.
I also noticed you use 'requestContext.getProperty( JaxRSEvent.NAME )' to retrieve the event. This is safe. Maybe we just use a new event every time?

Reply:
Yes it is simpler to create an event every time as you suggest. I did try running the tests to check the trace messages but my configuration of the gateway didn't work. I'll rework the commit to create a new instance of JaxRSEvent.